### PR TITLE
files: add a pager and a input filter on the files carousel

### PR DESCRIPTION
**Pull request by jma**
* Uses primeng bootstra4 theme to avoid layout problems.
* Adds a custom pager for the the files carousel.
* Adds an input to filter the files in the carousel.
* Makes the carousel reponsive.

### CODE REVIEW

File: projects/admin/src/app/record/detail-view/document-detail-view/files-collections/files-collections.component.ts [hunk 1]
```diff
@@ -69,6 +72,17 @@ export class FilesCollectionsComponent implements OnInit, OnDestroy {
...
 
+  /**
+   * Generate the public interface collection search link.
+   *
+   * @param name - the collection name
+   * @returns - url on the public interface
+   */
+  getCollectionLink(name): string {
+    const viewcode = this.organisationService.organisation.code;
+    return `https://localhost:5000/${viewcode}/search/documents?q=files.collections.raw:${name}&simple=0`;
```
Comment (lauren-d) [inline projects/admin/src/app/record/detail-view/document-detail-view/files-collections/files-collections.component.ts:83]: Why the URL is `https://localhost:5000` ?
Maybe you need to change that.

Comment (jma) [reply projects/admin/src/app/record/detail-view/document-detail-view/files-collections/files-collections.component.ts:83]: It was for debug. Thanks for the reporting.

File: projects/shared/src/lib/component/documents/files/files.component.html [hunk 2]
```diff
@@ -16,38 +16,114 @@
...
+        [showJumpToPageInput]="false"
+        [pageLinkSize]="3"
+      ></p-paginator>
+    }
+  }
 }
+
+
+<ng-template #previewModal>
+  <ng-container *ngIf="previewFile">
```
Comment (Garfield-fr) [inline projects/shared/src/lib/component/documents/files/files.component.html:104]: Use the new syntax `@if`. Check the file.

File: projects/shared/src/lib/component/documents/files/files.component.ts [hunk 3]
```diff
@@ -114,18 +166,22 @@ export class FilesComponent implements OnInit {
         })
       )
       .subscribe((res: any[]) => {
-        const files = [];
+        var files = [];
```
Comment (Garfield-fr) [inline projects/shared/src/lib/component/documents/files/files.component.ts:169]: `const`

**Review by Garfield-fr — APPROVED** [review]

**Pull request merged**